### PR TITLE
ASAN support, fix some crashes and refactor some globals

### PR DESCRIPTION
**Pull request by JJL772**
* Fix crash when terminals are configured without a bus coupler present 
* Move some globals into devEK9000 as statics
* Remove some unused code
* Add ASAN support

### CODE REVIEW

File: ek9000App/src/devEK9000.cpp [hunk 1]
```diff
@@ -888,7 +888,8 @@ void ek9000Configure(const iocshArgBuf* args) {
 	devEK9000* dev;
 
 	char ipbuf[64];
-	sprintf(ipbuf, "%s:%i", ip, port);
+	ipbuf[63] = 0;
+	snprintf(ipbuf, sizeof(ipbuf)-1, "%s:%i", ip, port);
```
Comment (klauer) [inline ek9000App/src/devEK9000.cpp]: I think [`snprintf`](https://cplusplus.com/reference/cstdio/snprintf/) buffer size accounts for the null terminator, but this is fine regardless

File: ek9000App/src/devEK9000.cpp [hunk 2]
```diff
@@ -945,7 +946,7 @@ void ek9000ConfigureTerminal(const iocshArgBuf* args) {
 
 void ek9000Stat(const iocshArgBuf* args) {
 	const char* ek9k = args[0].sval;
-	if (!ek9k || !g_pDeviceMgr) {
```
Comment (klauer) [inline ek9000App/src/devEK9000.cpp:948]: Was `g_pDeviceMgr` just not used? I guess that's point (3) in your PR description

File: ek9000App/src/devEK9000.h [hunk 3]
```diff
@@ -83,24 +83,21 @@
...
-extern bool g_bDebug;
-
 std::list<devEK9000*>& GlobalDeviceList();
 
 #define TERMINAL_FAMILY_ANALOG 0x1
 #define TERMINAL_FAMILY_DIGITAL 0x2
 
 #define DevInfo(fmt, ...)                                                                                              \
-	if (g_bDebug) {                                                                                                    \
+	if (devEK9000::debugEnabled) {                                                                                                    \
```
Comment (klauer) [inline ek9000App/src/devEK9000.h]: `clang-format` is not happy here according to CI... https://github.com/slaclab/epics-ek9000/actions/runs/3057306552/jobs/4932323207#step:3:42

File: ek9000App/src/devEK9000.h [hunk 4]
```diff
@@ -357,12 +354,11 @@ class devEK9000 {
...
 public:
-	/* Error reporting function, only prints on debug */
-	// void ReportError(int errorcode, const char* _msg = NULL);
-
-	/* Enable/disable debug */
-	// void EnableDebug(bool enabled) { m_debug = enabled; };
-
+	/* Statics! */
+	
+	static bool debugEnabled;
```
Comment (klauer) [inline ek9000App/src/devEK9000.h:359]: I like this style better - nice 👍

**Review by klauer — APPROVED** [review]
All looks good to me - just some style changes required for CI 👍

File: ek9000App/src/devEK9000.cpp (hunk 2 above)
Comment (JJL772) [reply ek9000App/src/devEK9000.cpp:948]: That is correct, g_pDeviceMgr was left over

File: ek9000App/src/devEK9000.cpp (hunk 1 above)
Comment (JJL772) [reply ek9000App/src/devEK9000.cpp]: You're right- I was thinking of strncpy in my mind when I wrote this

**Pull request merged**